store/postgres/src/store.rs

```diff
@@ -330,17 +330,58 @@ impl BasicStore for Store {
 }
 
 impl BlockStore for Store {
-    fn add_network_if_missing(&self, network_name: &str) -> Result<(), Error> {
+    fn add_network_if_missing(
+        &self,
+        new_network_name: &str,
+        new_net_version: &str,
+        new_genesis_block_hash: H256,
+    ) -> Result<(), Error> {
         use db_schema::ethereum_networks::dsl::*;
 
-        insert_into(ethereum_networks)
-            .values((
-                name.eq(network_name),
-                head_block_hash.eq::<Option<String>>(None),
-                head_block_number.eq::<Option<i64>>(None),
-            )).on_conflict(name)
-            .do_nothing()
-            .execute(&*self.conn.lock().unwrap())?;
+        let network_identifiers_opt = ethereum_networks
+            .select((net_version, genesis_block_hash))
+            .filter(name.eq(new_network_name))
+            .first::<(Option<String>, Option<String>)>(&*self.conn.lock().unwrap())
+            .optional()?;
+
+        match network_identifiers_opt {
+            // Network is missing in database
+            None => {
+                insert_into(ethereum_networks)
+                    .values((
+                        name.eq(new_network_name),
+                        head_block_hash.eq::<Option<String>>(None),
+                        head_block_number.eq::<Option<i64>>(None),
+                        net_version.eq::<Option<String>>(Some(new_net_version.to_owned())),
+                        genesis_block_hash
+                            .eq::<Option<String>>(Some(format!("{:x}", new_genesis_block_hash))),
+                    )).on_conflict(name)
+                    .do_nothing()
+                    .execute(&*self.conn.lock().unwrap())?;
+            }
+
+            // Network is in database and has identifiers
+            Some((Some(last_net_version), Some(last_genesis_block_hash))) => {
+                if last_net_version != new_net_version {
+                    panic!("Ethereum node provided net_version {}, but we expected {}. Did you change networks without changing the network name?", new_net_version, last_net_version);
```

Comment:
The two panic lines are a little too long. Can we break them up roughly like this?
```rust
panic!("Ethereum node provided net_version {}, \
        but we expected {}. Did you change networks \
        ...);
```

Reply:
Oh, good catch. I guess cargo fmt doesn't remove all of my formatting obligations :)